sdgx/data_models/metadata.py

```diff
@@ -147,41 +141,79 @@ def load(cls, path: str | Path) -> "Metadata":
         attributes = json.load(path.open("r"))
         return Metadata().update(attributes)
 
+    def check_single_primary_key(self, input_key: str):
+        """Check whether a primary key in column_list and has ID data type.
+
+        Args:
+            input_key(str): the input primary_key str
+        """
+
+        if input_key not in self.column_list:
+            raise MetadaError(f"Primary Key {input_key} not Exist in columns.")
+        if input_key not in self.id_columns:
+            raise MetadaError(f"Primary Key {input_key} should has ID DataType.")
+
+    def get_all_data_type_columns(self):
+        """Get all column names from `self.xxx_columns`.
+
+        All Lists with the suffix _columns in model fields and extend fields need to be collected.
+        All defined column names will be counted.
+
+        Returns:
+            all_dtype_cols(set): set of all column names.
+        """
+        all_dtype_cols = set()
+
+        # search the model fields and extend fields
+        for each_key in list(self.model_fields.keys()) + list(self._extend.keys()):
+            if each_key.endswith("_columns"):
+                column_names = self.get(each_key)
+                all_dtype_cols = all_dtype_cols.union(set(column_names))
+
+        return all_dtype_cols
+
     def check(self):
         """Checks column info.
 
         When passing as input to the next module, perform necessary checks, including:
-            -Is the primary key correctly defined.
-            -Is there any missing definition of the column.
-            -Are there any  unknown columns that have been incorrectly updated.
+            -Is the primary key correctly defined(in column list) and has ID data type.
+            -Is there any missing definition of each column in table.
+            -Are there any unknown columns that have been incorrectly updated.
         """
-        # Not implemented yet
+        # check primary key in column_list and has ID data type
+        for each_key in self.primary_keys:
+            self.check_single_primary_key(each_key)
 
-        pass
+        all_dtype_columns = self.get_all_data_type_columns()
 
-    def update_primary_key(self, primary_key: str | list[str], composite_primary_key: bool = False):
+        # check missing columns
+        for each_column in self.column_list:
+            if each_column not in all_dtype_columns:
+                raise MetadaError(f"Undefined data type for column {each_column}.")
+
+        # check unfamiliar columns in dtypes
+        for each_dtype_column in all_dtype_columns:
+            if each_dtype_column not in self.column_list:
+                raise MetadaError(f"Found undefined column: {each_dtype_column}.")
+
```

Comment:
Maybe we can use `set(self.column_list) - set(all_dtype_columns)` here, which we can indicate all invalid keys once.

Reply:
And maybe we can rename `MetadaError`  to `MetadataInvalidError`

Reply:
> Maybe we can use `set(self.column_list) - set(all_dtype_columns)` here, which we can indicate all invalid keys once.

This is a more concise way of writing.

The current implementation is because the error information raised in these two cases is different, which may make it easier for users to locate the two error types.

Reply:
> The current implementation is because the error information raised in these two cases is different, which may make it easier for users to locate the two error types.

There is no conflict, just use two if-expressions instead of two for-ifs.